cmd/soroban-cli/src/self_outdated_check.rs

```diff
@@ -0,0 +1,77 @@
+use crate::config::self_outdated_check::SelfOutdatedCheck;
+use crate::print::Print;
+use semver::Version;
+use serde::Deserialize;
+use std::error::Error;
+use std::time::Duration;
+
+const MINIMUM_CHECK_INTERVAL: Duration = Duration::from_secs(60 * 60 * 24); // 1 day
+const CRATES_IO_API_URL: &str = "https://crates.io/api/v1/crates/";
+const REQUEST_TIMEOUT: Duration = Duration::from_secs(30);
+
+#[derive(Deserialize)]
+struct CrateResponse {
+    #[serde(rename = "crate")]
+    crate_: Crate,
+}
+
+#[derive(Deserialize)]
+struct Crate {
+    #[serde(rename = "max_stable_version")]
+    max_stable_version: String,
+}
+
+/// Fetch the latest stable version of the crate from crates.io
+fn fetch_latest_stable_version() -> Result<String, Box<dyn Error>> {
+    let crate_name = env!("CARGO_PKG_NAME");
+    let url = format!("{CRATES_IO_API_URL}{crate_name}");
+    let response = ureq::get(&url).timeout(REQUEST_TIMEOUT).call()?;
+    let crate_data: CrateResponse = response.into_json()?;
+    Ok(crate_data.crate_.max_stable_version)
+}
+
+/// Print a warning if a new version of the CLI is available
+pub fn print_upgrade_prompt(quiet: bool) {
+    let current_version = env!("CARGO_PKG_VERSION");
```

Comment:
I'm inclined to get this from the function that we use elsewhere to contain where to get the version from, but I don't feel strongly the env var is fine.
```suggestion
    let current_version = crate::version::pkg();
```

Reply:
aeff349d18c88edb58bdd288abfe8c53696e19d6